website/docs/tutorials/08-testing/02-cheatcodes/expect-call.md

```diff
@@ -0,0 +1,64 @@
+# `expect_call`
+```python
+def expect_call(self, contract_address: int, fn_name: str, calldata: list[int]) -> None: ...
+```
+Allows to specify what calls to contracts' functions are expected in the code. You can use this cheatcode in any place in the code as it checks for the desired calls' existence at the very end of the program's execution.
+
+`calldata` is a list of arguments that are expected for a certain call. The checking mechanism goes for a strict match, which means that the order, as well as the values of each element, have to be exactly the same.
+
+
+```cairo title="Example"
+%lang starknet
+
+@contract_interface
+namespace MainContract {
+  func increase_balance(amount_1: felt, amount_2: felt, amount_3: felt) -> () {
+  }
+}
+
+@external
+func __setup__() {
+  %{
+    context.ctr_addr_a = deploy_contract("basic_contract.cairo", []).contract_address
+    context.ctr_addr_b = deploy_contract("basic_contract.cairo", []).contract_address
+  %}
+
+  return ();
+}
+
+
+@external
+func test_expect_call_success{syscall_ptr: felt*, range_check_ptr}() {
+  tempvar ctr_addr_a;
+  %{ ids.ctr_addr_a = context.ctr_addr_a %}
+  tempvar ctr_addr_b;
+  %{ ids.ctr_addr_b = context.ctr_addr_b %}
+  %{
+    expect_call(ids.ctr_addr_a, "increase_balance", [5, 6, 7])
+  %}
+
+  MainContract.increase_balance(contract_address=ctr_addr_a, amount_1=5, amount_2=6, amount_3=7);
+  MainContract.increase_balance(contract_address=ctr_addr_b, amount_1=1, amount_2=2, amount_3=3);
+
+  %{
+    expect_call(ids.ctr_addr_b, "increase_balance", [1, 2, 3])
+  %}
```

Comment:
We shouldn't encourage users to use this cheatcode below the actual call. I'd say it's a bug/limitation, not a feature.

```suggestion
    tempvar ctr_addr_a;
    tempvar ctr_addr_b;

    %{
        ids.ctr_addr_a = context.ctr_addr_a
        expect_call(ids.ctr_addr_a, "increase_balance", [5, 6, 7])
    %}
    MainContract.increase_balance(contract_address=ctr_addr_a, amount_1=5, amount_2=6, amount_3=7);

    %{
        ids.ctr_addr_b = context.ctr_addr_b
        expect_call(ids.ctr_addr_b, "increase_balance", [1, 2, 3])
    %}
    MainContract.increase_balance(contract_address=ctr_addr_b, amount_1=1, amount_2=2, amount_3=3);
```

Reply:
Sure, I think it's going to be more readable to group `expect_call`s together

Reply:
[don't show calling expect_call after the call in the docs](https://github.com/software-mansion/protostar/pull/1060/commits/974083f1225c6aa8ce800b2a53b94f927ba47dc2)